### DISCUSSION TITLE
Support "loose" locale detection

### DISCUSSION THREAD

**Discussion by alexturpin** [vercel/next.js#25600] [Ideas]
### Describe the feature you'd like to request

When using [Internationalized Routing](https://nextjs.org/docs/advanced-features/i18n-routing), if you are setting up an `fr` language it will not detect `fr-CA` as the French language and vice versa. I've found that here in Quebec, people's browsers can be configured with either `fr` or `fr-CA`, making it difficult to successfully detect their language with Next.js.

### Describe the solution you'd like

A `looseLocaleDetection` option in the i18n config block that would allow Next.js to match on the language only.

It seems as if Next.js uses [@hapi/accept](https://www.npmjs.com/package/@hapi/accept) to do the matching, which does not support such a feature. A similar module, [accept-language-parser](https://www.npmjs.com/package/accept-language-parser), does support it.

### Describe alternatives you've considered

One option is to add a bunch of derived locales in the next.config.js and have code in pages that handles all of them as one locale, but this is a bit of a messy approach.

Otherwise, short of a custom server, which limits deployment options, I can't think of other alternatives.

**Comment by alexturpin** [comment]
I would be happy to submit a PR if this is something that would be considered by the maintainers.

**Comment by olingern** [comment]
I'm also running into this problem. I would like `de` to match `/de-DE` and so forth. It would be nice if the `locales` array accepted an object where we could setup aliases. 

Thoughts @timneutkens ?

**Comment by jimhaakman** [comment]
Any update or (temporary) solutions for this? Running into this issue with nl-NL on Safari only.
